simulation/cmd/update_environment.py

```diff
@@ -0,0 +1,60 @@
+import os
+import argparse
+import json
+import datetime
+import time
+from dotenv import load_dotenv
+from simulation.utils import Query
+from simulation.utils.query import APIType
+from simulation.config import config_directory
+from simulation.common.race import Race, load_race
+
+
+load_dotenv()
+
+
+def main(race: str, weather_provider: str):
+    # Grab the race start date from settings
+    config_path = os.path.join(config_directory, f"settings_{race}.json")
+    with open(config_path) as f:
+        race_constants = json.load(f)
+
+    race = load_race(Race.RaceType(race))
+
+    date = datetime.datetime(race_constants["start_year"], race_constants["start_month"], race_constants["start_day"], 0, 0, 0)
+    race_begin_timestamp = time.mktime(date.timetuple())
+    current_timestamp = time.time()
+
+    # Check if the current time makes sense (within race duration) and abort if not
+    time_difference: float = current_timestamp - race_begin_timestamp  # Units: Seconds
+
+    if time_difference < 0 or time_difference > race.race_duration:
+        raise ValueError(f"Current time does not make sense for the current context. Is {date} the correct start date for {race}?")
+
+    # Set the time (timezone-dependent UNIX time) in config
+    initial_conditions_path = os.path.join(config_directory, f"initial_conditions_{race}.json")
+    with open(initial_conditions_path, 'r') as file:
+        initial_conditions = json.load(file)
+
+    initial_conditions['start_time'] = int(time_difference)
```

Comment:
just wondering, why is start_time being updated to the time difference?

Reply:
`time_difference` is going to be the number of seconds since the start of the race, since we find the current time in UNIX time and subtract off the start time of the race in UNIX time. That's what we want to be in `start_time` which defines the real time that Simulation thinks its being run at (and for weather caching, too).

Reply:
Sounds good, just wanted to confirm. Optionally we could change the variable names to be a bit more descriptive, but up to you if you think its necessary:
start_time  -> simulation_start_time
time_difference -> elapsed_time

Reply:
I agree, I'll make those changes.